engine/gameobject/src/gameobject/gameobject_script.cpp

```diff
@@ -623,18 +635,45 @@ namespace dmGameObject
         {
         case dmGameObject::PROPERTY_RESULT_OK:
             {
-                if (index_requested && !property_desc.m_IsArray)
```

Comment:
this one

Reply:
I don't see a reason that this new if-statement would fail, just because ARRAY=0 and HASHTABLE=1 ?

Reply:
In café when it is not an array and not a hash table but user tries to specify index

Reply:
If I keep one byte I need to add array length in this structure again, and use it instead

Reply:
>In case when it is not an array and not a hash table but user tries to specify index

But as we discussed previously, it can only be an array _or_ a hashtable.
See the discussion about the property type

Reply:
```
    enum PropertyValueType
    {
        PROP_VALUE_HASHTABLE,
        PROP_VALUE_ARRAY    
    };
 ```

Reply:
In our discussion we said that a "variable" is an array of size 1.